document/FileFormat.mdk

```diff
@@ -2408,23 +2412,132 @@ This allows tools to include information about the stack frame that is not expli
 An `annotatedCodeLocation` object represents a physical location together with additional information
 relevant to the use of the location in a particular context.
 
+### `id` property { #annotatedCodeLocation-id }
+
+An `annotatedCodeLocation` object may contain a property named `id` whose value is a string that provides
+an identifier for the location. This identifier shall be unique within the scope of the `codeFlow` object (&sect;[#codeFlow])
+within which this `annotatedCodeLocation` occurs.
+If this `annotatedCodeLocation` does not occur within a `codeFlow`, the `id` property shall be absent.
+
+~ IsoNumberedNote { @IsoNumberedNote=1 }
+The primary purpose of this property is to allow a viewer to display the identifer when it displays
+the list of locations that comprise a code flow.
+This enables users to easily refer to the location in conversation, for example,
+"I think the problem occurs at location 6".
+~
+
+~ IsoNumberedNote
+The `message` property (&sect;[#annotatedCodeLocation-message]) of an `annotatedCodeLocation` can use the `id` property
+to refer to another `annotatedCodeLocation` within the same `codeFlow` object.
```

Comment:
I'm not sure I like this. Interesting, Jakob said the ids weren't used in this manner. Providing this guidance in the spec arguably puts a demand on viewers to display these ids (otherwise messages can't meaningfully refer to them). For this kind of use, why isn't the index of the array sufficient as an id? I guess I suggest starting a thread with Jakob on this topic.